docs/sphinx_doc/user/inputs.rst

```diff
@@ -159,14 +163,530 @@ This section deals with parameters that control the simulation.
+   This maybe specified as a single number for all levels or a value per AMR level.
+   
+.. input_param:: incflo.gradrhoerr
+
+   **type:** Real number or a list of Real numbers
+
+   When `amr.max_level > 0` this will trigger mesh adaption if the difference 
+   between density at a cell center and its neighbors is greater than `incflo.gradrhoerr`. 
+   This maybe specified as a single number for all levels or a value per AMR level.
+   
+Section: ``transport``
+~~~~~~~~~~~~~~~~~~~~~~
+
+This section is for setting thermal and momentum diffusivities.
+
+.. input_param:: transport.model
+
+   **type:** String, optional, default = ConstTransport
+
+   Currently this is the only transport model implemented.
+   
+.. input_param:: transport.viscosity
+
+   **type:** Real, optional, default = 1.0e-5
+
+   Sets the dynamic viscosity
+   
+.. input_param:: transport.laminar_prandtl 
+
+   **type:** Real, optional, default = 1.0
+
+   Sets the laminar Prandtl number.
+   
+.. input_param:: transport.turbulent_prandtl 
+
+   **type:** Real, optional, default = 1.0
+
+   Sets the turbulent Prandtl number.
+   
+Section: ``turbulence``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+This section is for setting turbulence model parameters
+
+.. input_param:: turbulence.model
+
+   **type:** String, optional, default = Laminar
+
+   Specifies which turbulence model to use, by default ``Laminar`` is chosen 
+   (effectively no turbulence model). 
+   Currently the only turbulence model is ``Smagorinsky``
+
+   
+.. input_param:: Smagorinsky_coeffs.Cs
+
+   **type:** Real, optional, default = 0.135
+
+   Specifies the coefficient used in the Smagorinsky turbulence model. 
+   
+Section: ``ABL``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+This section is for setting atmospheric boundary layer parameters.
+
+.. input_param:: abl.kappa
+
+   **type:** Real, optional, default = 0.41
+   
+   Wall model coefficient.
+   
+.. input_param:: abl.surface_roughness_z0
+
+   **type:** Real, optional, default = 0.1
+   
+   Wall model surface roughness. 
+   
+.. input_param:: abl.normal_direction
+
+   **type:** Integer, optional, default = 2
+   
+   Wall model normal direction. x-direction = 0, y-direction = 1, z-direction = 2.
+
+.. input_param:: abl.log_law_height
+
+   **type:** Real, optional
+   
+   Height to evaluate the log law for the wall model. 
+   Currently, if this parameter is not specified in the input file, the first half cell height is calculated
+   and is used to set the log law height. 
+   Therefore the log law height depends on the domain size and number of elements in the normal direction. 
+   If this parameter is set to a number the log law is evaluated at a fixed height. 
+   Note: currently the fluctuating velocity terms in the shear stress model are only 
+   available at the first cell center above the wall. This limitation will be removed soon.
+   
+.. input_param:: abl.ntemperature
```

Comment:
Technically `ParmParse` doesn't need this. It will automatially resize the `amrex::Vector`, we should probably add a check to ensure that `temperature_heights` and `temperature_values` are the same length.

Reply:
I was worried about the 0 length vectors inside the field initialization. But maybe those are ok too. I'll check